datafusion/core/src/execution/memory_manager/mod.rs

```diff
@@ -17,419 +17,187 @@
 
 //! Manages all available memory during query execution
 
-use crate::error::{DataFusionError, Result};
-use async_trait::async_trait;
-use hashbrown::HashSet;
-use log::{debug, warn};
-use parking_lot::{Condvar, Mutex};
-use std::fmt;
-use std::fmt::{Debug, Display, Formatter};
-use std::sync::atomic::{AtomicUsize, Ordering};
+use crate::error::Result;
 use std::sync::Arc;
-use std::time::{Duration, Instant};
 
+mod pool;
 pub mod proxy;
 
-static CONSUMER_ID: AtomicUsize = AtomicUsize::new(0);
-
-#[derive(Debug, Clone)]
-/// Configuration information for memory management
-pub enum MemoryManagerConfig {
```

Comment:
This was a bit of a perverse API imo, I'm looking into properly addressing the `*Config` profileration in #4349 but I figured I might as well clean this up whilst I was here.

If people want to reuse the same pool, they can just use the same `Arc<dyn MemoryPool>`